app/Http/Controllers/PortGroupController.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\PortGroup;
+use Illuminate\Http\Request;
+use Illuminate\Validation\Rule;
+use Toastr;
+
+class PortGroupController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        return view('port-group.index', [
+            'port_groups' => PortGroup::orderBy('name')->get(),
+        ]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('port-group.create', [
+            'port_group' => new PortGroup(),
+        ]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, [
+            'name' => 'required|string|unique:port_groups',
+        ]);
+
+        $portGroup = PortGroup::make($request->only(['name', 'desc']));
+        $portGroup->save();
+
+        Toastr::success(__('Port Group :name created', ['name' => $portGroup->name]));
+
+        return redirect()->route('port-groups.index');
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param \App\Models\PortGroup $portGroup
+     * @return \Illuminate\Http\Response
+     */
+    public function edit(PortGroup $portGroup)
+    {
+        return view('port-group.edit', [
+            'port_group' => $portGroup,
+        ]);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \App\Models\PortGroup $portGroup
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, PortGroup $portGroup)
+    {
+        $this->validate($request, [
+            'name' => [
+                'required',
+                'string',
+                Rule::unique('port_groups')->where(function ($query) use ($portGroup) {
+                    $query->where('id', '!=', $portGroup->id);
+                }),
```

Comment:
This shouldnt be here I think, if we update a port group we shouldnt check if its unique?

Reply:
therefore a validater is here?

Reply:
you check if the id doesnt exist, when you try to update it, so it will always fail to update due to it already existing?

Reply:
i added/updated name and/or description of a Port Group in PortGroup Management and it worked